### PR TITLE
Fixes #105, move the fqdn_resolves method

### LINKED ISSUE #105: move helpers to chef-ingredient cookbook
The [chef-ingredient](https://supermarket.chef.io/cookbooks/chef-ingredient) is a more appropriate place for these primitives, as they will also be necessary for other "chef server" type cookbooks.

### PR DESCRIPTION
**Pull request by jtimberman**
The api_fqdn_resolves? method should use the new method that is in the chef-ingredient cookbook. It is implemented in that cookbook here:

- https://github.com/chef-cookbooks/chef-ingredient/commit/14a54e9dfa10333b666aa9c90947c9e6f4a86e73

Reference:

- https://github.com/chef-cookbooks/chef-ingredient/pull/36
- https://github.com/chef-cookbooks/chef-ingredient/issue/35

### CODE REVIEW

**Comment by jtimberman** [comment]
@chef-cookbooks/engineering-services

File: libraries/helpers.rb [hunk 1]
```diff
@@ -8,11 +8,8 @@ def api_fqdn_available?
     end
 
     def api_fqdn_resolves?
-      require 'resolv'
-      Resolv.getaddress(node['chef-server']['api_fqdn'])
-      return true
-    rescue
-      false
+      extend ChefIngredientCookbook::Helpers
+      fqdn_resolves?(node['chef-server']['api_fqdn'])
```
Comment (schisamo) [inline libraries/helpers.rb]: See [this comment](https://github.com/chef-cookbooks/chef-ingredient/pull/36/files#r38191000) about using `module_function` to make these sorts of method calls less awkward.

**Comment by patrick-wright** [comment]
:+1:

**Comment by schisamo** [comment]
:+1:

**Pull request merged**